samples/smp/pi/src/main.c

```diff
@@ -17,7 +17,8 @@
 #define DIGITS_NUM	240
 
 #define LENGTH		((DIGITS_NUM / 4) * 14)
-#define STACK_SIZE	(LENGTH * sizeof(int) + 512)
+#define STACK_SIZE	((LENGTH * sizeof(int) + 512) + \
+                         CONFIG_TEST_EXTRA_STACKSIZE)
```

Comment:
That's weird. Either I made calculations on stack size requirements fundamentally wrong or there should be something else. See I already added extra 512 bytes to buffer size being used in calculations and in the worker thread itself there seem to be not many allocations made on stack as well as only `sprintf()` is explicitly called and probably some math ops are done via libgcc's implementations.

Anyways I guess we should:
1. **Replace** hardcoded `512` with `CONFIG_TEST_EXTRA_STACKSIZE` which is 4096 for `x86_64`.
2. Figure out how much stack is being used by Zephyr's code running in thread's context but excluding real body of the worker thread.

Reply:
sprintf() uses a lot of stack. on some arches 512 isn't enough, which is why for all tests/samples we add CONFIG_TEST_EXTRA_STACKSIZE to thread stack sizes.

Reply:
But given we add 4096 bytes with `CONFIG_TEST_EXTRA_STACKSIZE`, do we need those hardcoded 512 bytes?

Reply:
a lot of platforms have CONFIG_TEST_EXTRA_STACKSIZE as 0, so yes I think we do

Reply:
Ok then, agree.